src/DonationForms/AsyncData/resources/loadAsyncData.js

```diff
@@ -0,0 +1,302 @@
+/**
+ * This file contains the whole necessary logic to load async data on all available form list views
+ * in the project - form grid and admin form list views. The async data are loaded (only for the
+ * items visible on the screen) at the page's first load and every time the user scrolls the mouse.
+ *
+ * @unreleased
+ */
+document.addEventListener('DOMContentLoaded', () => {
+    /**
+     * We are declaring it at the top to use it in more than one function.
+     */
+    let abortLoadAsyncData = false;
+    const giveListTable = document.querySelector('.giveListTable');
+    const giveListTableIsLoadingEvent = new Event('giveListTableIsLoading');
+
+    /**
+     * This function check if the element is visible on the screen.
+     *
+     * @unreleased
+     */
+    function isInViewport(element) {
+        const {top, bottom} = element.getBoundingClientRect();
+        const vHeight = window.innerHeight || document.documentElement.clientHeight;
+
+        return (top > 0 || bottom > 0) && top < vHeight;
+    }
+
+    /**
+     * Check if an element is a placeholder waiting to have the value updated.
+     *
+     * @unreleased
+     */
+    function isPlaceholder(element) {
+        return !!element && Boolean(element.querySelector('.js-give-async-data'));
+    }
+
+    /**
+     * This function fetch the async data from the server and set the values to the proper elements in the DOM.
+     *
+     * @unreleased
+     */
+    const loadFormData = (
+        formId,
+        itemElement,
+        amountRaisedElement = null,
+        progressBarElement = null,
+        goalAchievedElement = null,
+        donationsElement = null,
+        earningsElement = null
+    ) => {
+        // If we don't have any of these elements with a placeholder waiting to be updated, then return.
+        if (
+            !isPlaceholder(amountRaisedElement) &&
+            !isPlaceholder(donationsElement) &&
+            !isPlaceholder(earningsElement)
+        ) {
+            return;
+        }
+
+        window.GiveDonationFormsAsyncData.scriptDebug && console.log('item: ', itemElement);
+
+        // This class ensures that once the element has the fetch request triggered we'll not try to fetch it again.
+        itemElement.classList.add('give-async-data-fetch-triggered');
+
+        url =
+            window.GiveDonationFormsAsyncData.ajaxUrl +
+            '?action=givewp_get_form_async_data_for_list_view&formId=' +
+            formId +
+            '&nonce=' +
+            window.GiveDonationFormsAsyncData.ajaxNonce;
```

Comment:
Simplify a little bit, you could also build this url largely in php as well

```suggestion
        const url = `${window.GiveDonationFormsAsyncData.ajaxUrl}?action=givewp_get_form_async_data_for_list_view&formId=${formId}&nonce=${window.GiveDonationFormsAsyncData.ajaxNonce}`;
```